bar.go

```diff
@@ -95,6 +95,10 @@ func (b *Bar) Set(n int) error {
 	if n > b.Total {
 		return ErrMaxCurrentReached
 	}
+	if b.TimeStarted.IsZero() {
+		b.TimeStarted = time.Now()
+	}
```

Comment:
Would you consider the alternative of using implementing `Incr()` by reusing `Set()`? This would result in more reused code and less code overall, less opportunity for error.

Ideally there would be a test as well.